### PR TITLE
PARQUET-2425: Support non-grouped repeated fields in AvroSchemaConverter

### PR DESCRIPTION
**Pull request by clairemcginty**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references
  them in the PR title. For example, "PARQUET-1234: My Parquet PR"
    - https://issues.apache.org/jira/browse/PARQUET-2425
    - In case you are adding a dependency, check if the license complies with
      the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines
  from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    1. Subject is limited to 50 characters (not including Jira issue reference)
    1. Subject does not end with a period
    1. Subject uses the imperative mood ("add", not "adding")
    1. Body wraps at 72 characters
    1. Body explains "what" and "why", not "how"

### Style
- [x] My contribution adheres to the code style guidelines and Spotless passes.
    - To apply the necessary changes, run `mvn spotless:apply -Pvector-plugins`

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Review by wgtmac — APPROVED** [review]
+1

Thanks @clairemcginty!

**Comment by clairemcginty** [comment]
thanks! @Fokko , does this look ok to you?

**Comment by wgtmac** [comment]
cc @gszadovszky

**Review by Fokko — APPROVED** [review]
I think it is reasonable to support lists and maps as well

**Review by gszadovszky — APPROVED** [review]

File: parquet-avro/src/main/java/org/apache/parquet/avro/AvroSchemaConverter.java [hunk 1]
```diff
@@ -300,9 +300,8 @@ private Schema convertFields(String name, List<Type> parquetFields, Map<String,
     Integer nameCount = names.merge(name, 1, (oldValue, value) -> oldValue + 1);
     for (Type parquetType : parquetFields) {
       Schema fieldSchema = convertField(parquetType, names);
-      if (parquetType.isRepetition(REPEATED)) {
-        throw new UnsupportedOperationException(
-            "REPEATED not supported outside LIST or MAP. Type: " + parquetType);
+      if (parquetType.isRepetition(REPEATED)) { // If a repeated field is ungrouped, treat as REQUIRED per spec
+        fields.add(new Schema.Field(parquetType.getName(), Schema.createArray(fieldSchema)));
```
Comment (RustedBones) [inline parquet-avro/src/main/java/org/apache/parquet/avro/AvroSchemaConverter.java:304]: This constructor is not available in avro 1.8. This leads to a runtime error
```
Cause: java.lang.NoSuchMethodError: 'void org.apache.avro.Schema$Field.<init>(java.lang.String, org.apache.avro.Schema)'
```
When using `parquet-avro` 1.14 with avro version 1.8

**Pull request merged**